### DISCUSSION TITLE
After upgrade - HASPone shows blank pages

### DISCUSSION THREAD

**Discussion by douglasmauro** [HASwitchPlate/HASPone#195] [Q&A] [answered]
After some typical updates my plate stopped working.  It appears the configured pages (empty boxes) still show when I press the lower buttons but nothing else shows anymore .. icons, temp, etc.  It's just blank.  I've restarted and rebooted anything I could.

I've tried going into the scripts and fire an event (on button press) 

.... these msgs continue until it fills screen ....
error: received null username, clientid or topic, or access is equal or less than 0 for acl check
error: received null username, clientid or topic, or access is equal or less than 0 for acl check
error: received null username, clientid or topic, or access is equal or less than 0 for acl check
error: received null username, clientid or topic, or access is equal or less than 0 for acl check
error: received null username, clientid or topic, or access is equal or less than 0 for acl check
error: received null username, clientid or topic, or access is equal or less than 0 for acl check
1655735136: Client <unknown> disconnected due to protocol error.
1655735136: New connection from 192.168.86.68:59904 on port 1883.
1655735136: New client connected from 192.168.86.68:59904 as plate01-8cce4ed5a379 (p2, c1, k30, u'mymqtt').

Home Assistant Core 2022.6.6
Home Assistant Supervisor 2022.05.3
Home Assistant OS 8.1

MQTT Status: Connected
MQTT ClientID: plate01-8cce4ed5a379
HASPone FW Version: 1.05
LCD Model: NX3224T024_011R
LCD FW Version: 3
LCD Active Page: 6
LCD Serial Speed: 921600

**Comment by aderusha** [comment]
Does the web interface on the HASPone device show you any information?  Toward the bottom you'll find some details about what it thinks its doing.  Next, dumping some debug logs somewhere where we can review them together would be helpful, more on that here: https://github.com/HASwitchPlate/HASPone/wiki/Capture-HASPone-debug-output

**Reply by douglasmauro** [reply]
I can login to the web interface and get the usual info.  No issues that I can see.  I tried an upgrade as well but it didn't seem to up my FW version.

MQTT Status: Connected
MQTT ClientID: plate01-8cce4ed5a379
HASPone FW Version: 1.05
LCD Model: NX3224T024_011R
LCD FW Version: 3
LCD Active Page: 9
LCD Serial Speed: 921600
CPU Frequency: 160MHz
Sketch Size: 661728 bytes
Free Sketch Space: 2482176 bytes
Heap Free: 18648
Heap Fragmentation: 13
ESP core version: 3.0.2
IP Address: 192.168.86.68
Signal Strength: -59
Uptime: 197
Last reset: Software/System restart

I'm going in now to try the telnet and report back here with details.  THANKS!

**Reply by douglasmauro** [reply]
Going into scripts (that have worked for months now no issue) don't seem to be working (which updates the screen icons/data/etc)  I manually run the scripts and don't see anything in telnet nor from mqtt explorer.  When I push the 3 buttons on bottom it does show data in both (and the screens do change).

Here's the telnet debug.  I am simply clicking one of the bottom icons to switch pages.  The pages DO switch but nothing shows up in the boxes:

[+60.001s] UPDATE: Checking update URL: https://raw.githubusercontent.com/HASwitchPlate/HASPone/main/update/version.json
[+60.884s] UPDATE: updateEspAvailableVersion: 1.05
[+60.886s] UPDATE: updateLcdAvailableVersion: 3
[+60.887s] UPDATE: Update check completed
[+60.920s] MQTT OUT: 'hasp/plate01/sensor' : '{"espVersion":1.05,"updateEspAvailable":false,"lcdConnected":true,"lcdVersion":"3","updateLcdAvailable":false,"espUptime":60,"signalStrength":-61,"haspName":"plate01","haspIP":"192.168.86.68","haspClientID":"plate01-8cce4ed5a379","haspMac":"8c:ce:4e:d5:a3:79","haspManufacturer":"HASwitchPlate","haspModel":"HASPone v1.0.0","heapFree":18168,"heapFragmentation":32,"heapMaxFreeBlockSize":9128,"espCore":"3.0.2"}'
[+77.710s] HMI OUT:
[+78.711s] HMI ERROR: Nextion Ack timeout
[+78.713s] MQTT OUT: 'hasp/plate01/state/json' : '{"event":"nextionError","value":"Nextion Ack timeout"}'
[+84.173s] HMI IN: [7 bytes]: 0x65 0x09 0x02 0x01 0xff 0xff 0xff
[+84.175s] HMI IN: [Button ON] 'p[9].b[2]'
[+84.176s] MQTT OUT: 'hasp/plate01/state/p[9].b[2]' : 'ON'
[+84.193s] MQTT OUT: 'hasp/plate01/state/json' : '{"event_type":"button_short_press","event":"p[9].b[2]","value":"ON"}'
[+84.396s] HMI IN: [7 bytes]: 0x65 0x09 0x02 0x00 0xff 0xff 0xff
[+84.398s] HMI IN: [Button OFF] 'p[9].b[2]'
[+84.399s] MQTT OUT: 'hasp/plate01/state/p[9].b[2]' : 'OFF'
[+84.416s] MQTT OUT: 'hasp/plate01/state/json' : '{"event_type":"button_short_release","event":"p[9].b[2]","value":"OFF"}'
[+84.421s] HMI OUT: 'get p[9].b[2].val'
[+84.422s] HMI IN: [8 bytes]: 0x71 0x00 0x00 0x00 0x00 0xff 0xff 0xff
[+84.430s] HMI IN: [Int Return] '0'
[+84.437s] MQTT OUT: 'hasp/plate01/state/p[9].b[2].val' : '0'
[+84.446s] MQTT OUT: 'hasp/plate01/state/json' : '{"event":"p[9].b[2].val","value":0}'
[+84.487s] MQTT IN: 'hasp/plate01/command/page' : '2'
[+84.489s] HMI OUT: page 2
[+84.549s] HMI IN: [5 bytes]: 0x66 0x02 0xff 0xff 0xff
[+84.550s] HMI IN: [sendme Page] '2'
[+84.551s] MQTT OUT: 'hasp/plate01/state/json' : '{"event":"page","value":2}'
[+84.570s] MQTT OUT: 'hasp/plate01/state/page' : '2'
[+94.514s] HMI IN: [7 bytes]: 0x65 0x02 0x03 0x01 0xff 0xff 0xff
[+94.516s] HMI IN: [Button ON] 'p[2].b[3]'
[+94.518s] MQTT OUT: 'hasp/plate01/state/p[2].b[3]' : 'ON'
[+94.545s] MQTT OUT: 'hasp/plate01/state/json' : '{"event_type":"button_short_press","event":"p[2].b[3]","value":"ON"}'
[+94.678s] HMI IN: [7 bytes]: 0x65 0x02 0x03 0x00 0xff 0xff 0xff
[+94.680s] HMI IN: [Button OFF] 'p[2].b[3]'
[+94.681s] MQTT OUT: 'hasp/plate01/state/p[2].b[3]' : 'OFF'
[+94.702s] MQTT OUT: 'hasp/plate01/state/json' : '{"event_type":"button_short_release","event":"p[2].b[3]","value":"OFF"}'
[+94.703s] HMI OUT: 'get p[2].b[3].val'
[+94.705s] HMI IN: [8 bytes]: 0x71 0x00 0x00 0x00 0x00 0xff 0xff 0xff
[+94.726s] HMI IN: [Int Return] '0'
[+94.728s] MQTT OUT: 'hasp/plate01/state/p[2].b[3].val' : '0'
[+94.729s] MQTT OUT: 'hasp/plate01/state/json' : '{"event":"p[2].b[3].val","value":0}'
[+94.757s] MQTT IN: 'hasp/plate01/command/page' : '6'
[+94.760s] HMI OUT: page 6
[+94.821s] HMI IN: [5 bytes]: 0x66 0x06 0xff 0xff 0xff
[+94.823s] HMI IN: [sendme Page] '6'
[+94.824s] MQTT OUT: 'hasp/plate01/state/json' : '{"event":"page","value":6}'
[+94.843s] MQTT OUT: 'hasp/plate01/state/page' : '6'

**Reply by aderusha** [reply]
Allright, thanks for that!  Based on that log, the HASPone is happy and doing its thing.  So, problem is on the Home Assistant side and I bet I know what's going on.  Try the process outlined here: https://github.com/HASwitchPlate/HASPone/wiki/Fix-common-HASPone-issues

**Reply by douglasmauro** [reply]
No orphans and did reset.  Still nothing fires.  I am able to change pages via HA plate page no problem so it appears HA can talk with plate fine.  It's the scripts and automations that are not working.

Also checked that each of the scripts have the plate01.

**Reply by douglasmauro** [reply]
I've also created a brand new auto from blueprint (simple toggle button) and that didn't work either.  Same results.

**Reply by douglasmauro** [reply]
Looking more into logs I am seeing a HUGE amount of these:

Log Details (ERROR)
Logger: homeassistant.components.automation.haspone_p_x_b_y_displays_the_month_date_with_a_calendar_icon
Source: components/automation/__init__.py:470
Integration: Automation ([documentation](https://www.home-assistant.io/integrations/automation), [issues](https://github.com/home-assistant/home-assistant/issues?q=is%3Aissue+is%3Aopen+label%3A%22integration%3A+automation%22))
First occurred: 12:04:26 PM (318 occurrences)
Last logged: 12:19:27 PM

Error rendering variables: UndefinedError: None has no element 0

Error rendering variables: UndefinedError: None has no element 0
12:19:42 PM – (ERROR) Automation - message first occurred at 12:18:48 PM and shows up 154 times
Error rendering variables: UndefinedError: None has no element 0
12:19:27 PM – (ERROR) Automation - message first occurred at 12:04:26 PM and shows up 266 times
Error rendering variables: UndefinedError: None has no element 0
12:19:27 PM – (ERROR) Automation - message first occurred at 12:04:26 PM and shows up 266 times
Error rendering variables: UndefinedError: None has no element 0
12:19:27 PM – (ERROR) Automation - message first occurred at 12:04:26 PM and shows up 370 times
Error rendering variables: UndefinedError: None has no element 0
12:19:27 PM – (ERROR) Automation - message first occurred at 12:04:26 PM and shows up 266 times
Error rendering variables: UndefinedError: None has no element 0
12:19:27 PM – (ERROR) Automation - message first occurred at 12:04:26 PM and shows up 318 times
Error rendering variables: UndefinedError: None has no element 0
12:19:27 PM – (ERROR) Automation - message first occurred at 12:04:26 PM and shows up 266 times
Error rendering variables: UndefinedError: None has no element 0
12:19:27 PM – (ERROR) Automation - message first occurred at 12:04:26 PM and shows up 266 times
Error rendering variables: UndefinedError: None has no element 0
12:19:27 PM – (ERROR) Automation - message first occurred at 12:04:26 PM and shows up 266 times
Error rendering variables: UndefinedError: None has no element 0
12:19:27 PM – (ERROR) Automation - message first occurred at 12:04:26 PM and shows up 266 times

**Reply by aderusha** [reply]
Ahh... OK I think what's happening is that you have old versions of the blueprints installed.  Here's the annoying bit, Home Assistant offers no automated (or even easily-accessible) process for updating a blueprint.  If you have a minute, one thing that can help this process is to upvote this feature request: https://community.home-assistant.io/t/allow-blueprint-upgrades/366939

OK so, what you're probably going to need to do here is to delete the existing blueprint file from your installation, and [re-run the import process](https://github.com/HASwitchPlate/HASPone/wiki/HASPone-Blueprints-for-Home-Assistant).

**Reply by aderusha** [reply]
To remove blueprints, you'll need to find the `automation/blueprints/HASwitchPlate` folder in your Home Assistant config folder, delete those, then re-import from the repository: https://github.com/HASwitchPlate/HASPone/tree/main/Home_Assistant/blueprints#haspone-blueprint-exchange

**Reply by douglasmauro** [reply]
Will I need to remove all my plate automations and re-do them again with these new blueprints or just have to delete and re import the new blueprints?

Thanks!

**Reply by douglasmauro** [reply]
After deleting (mv/safe) the old blueprints, It appears that I've lost all my automations now with the plate.  I did confirm that a new automation from a new blueprint does work!  But I need to figure out how to get all my settings back.  Ideas??

**Reply by aderusha** [reply]
After deleting the old blueprints, you need to re-import them again.  Your automations will not work until the blueprint has been replaced.  So, simply go through the import process for each blueprint, and your existing automations should start working again (possibly after a restart of home assistant).

**Comment by douglasmauro** [comment, marked as answer]
**UPDATE:  WORKS!  Just needed to fire off the HASPone Core functionality once (added in steps below)**

Here are the steps I took

rm all plate BP files
reimport same BP files one at a time
verify file names matched prev BP files
restart HA
within BP I see the automations but they still don't work.
NEW:  Go into HASPone Core functionality - First-time HASPone setup or reset / save / run / uncheck / save

WORKS!!!

If I open any of them I see "Blueprint Version: 1.05.00" and I even tried
changing some things to save/run with no results.

Logging still showing a ton of:

Logger:
homeassistant.components.automation.haspone_p_x_b_y_displays_a_clock_with_a_clock_icon
Source: components/automation/__init__.py:470
Integration: Automation (documentation, issues)
First occurred: 11:16:03 AM (237 occurrences)
Last logged: 11:28:00 AM

Error rendering variables: UndefinedError: None has no element 0


Douglas

**Reply by aderusha** [reply]
Can you take one of those automations, open it in the automation editor, click the 3 dots on the top right, and say "edit in YAML", then paste the code you see there here in the thread?  

Here's an example from one of my devices:

```yaml
alias: deskhasp p[1].b[4] displays a clock with a clock icon
description: ''
use_blueprint:
  path: HASwitchPlate/hasp_Display_Clock_with_Icon.yaml
  input:
    haspdevice: 4f4df4aece36670a3ed90624850a5310
```

**Reply by douglasmauro** [reply]
(not sure if we posted at same time ... fixed with that core run!)  many thanks!!

```yaml
alias: HASPone p[2].b[4] displays a clock with a clock icon
description: ''
use_blueprint:
  path: HASwitchPlate/hasp_Display_Clock_with_Icon.yaml
  input:
    haspdevice: d950784be9172c5de62727bceeceeecb
    hour24: false
    hasppage: 2
```

**Reply by aderusha** [reply]
Fantastic!  Everything working now then?

**Reply by douglasmauro** [reply]
Sure is and looks like those errors went away too!  Many thanks for all the help!  Felt so good to see everything pop back up on my screen.
